src_c/event.c

```diff
@@ -1960,6 +1960,112 @@ _pg_event_append_to_list(PyObject *list, SDL_Event *event)
     return 1;
 }
 
+static PyObject *
+_pg_get_all_events_except(PyObject *obj)
+{
+    SDL_Event event;
+    int loop, type, len, ret;
+    PyObject *seq, *list;
+
+    SDL_Event *filtered_events;
+    int filtered_index = 0;
+    int filtered_events_len = 16;
+
+    SDL_Event eventbuf[PG_GET_LIST_LEN];
+
+    filtered_events=malloc(sizeof(SDL_Event)*filtered_events_len);
+    if (!filtered_events)
+        return PyErr_NoMemory();
+
+    list = PyList_New(0);
+    if (!list)
+        return PyErr_NoMemory();
+
+    seq = _pg_eventtype_as_seq(obj, &len);
+    if (!seq)
+        goto error;
+
+    for (loop = 0; loop < len; loop++) {
+        type = _pg_eventtype_from_seq(seq, loop);
+        if (type == -1)
+            goto error;
+
+        do {
+            ret = PG_PEEP_EVENT(&event, 1, SDL_GETEVENT, type);
+            if (ret < 0) {
+                PyErr_SetString(pgExc_SDLError, SDL_GetError());
+                goto error;
+            }
+            else if (ret > 0) {
+                if (filtered_index==filtered_events_len) {
+                    SDL_Event *new_filtered_events = malloc(sizeof(SDL_Event)*filtered_events_len*4);
```

Comment:
Sorry, I'm a bit daft... Why is this *4 here?

Reply:
Growing the allocated array. Could also be filtered_events_len*2 or filtered_events_len*3, or even filtered_events_len+1.